js/src/controls/WidgetControl.js

```diff
@@ -0,0 +1,83 @@
+var widgets = require('@jupyter-widgets/base');
+var _ = require('underscore');
+var L = require('../leaflet.js');
+var control = require('./Control.js');
+var LeafletControlView = control.LeafletControlView;
+var LeafletControlModel = control.LeafletControlModel;
+
+L.Control.WidgetControl = L.Control.extend({
+    options: {
+        position: 'topright',
+    },
+
+    _updateContent: function () {
+		if (!this._content) { return; }
+        var node = this._container
+        L.DomEvent.disableClickPropagation(node);
+        L.DomEvent.disableScrollPropagation(node);
+        var content = this._content;
+		while (node.hasChildNodes()) {
+			node.removeChild(node.firstChild);
+		}
```

Comment:
That is not a clean way to remove old widget views. You need to call `view.remove()` on the old view if it exists.